extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/config/TransitKeyConfig.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.vault.runtime.config;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class TransitKeyConfig {
+
+    /**
+     * Specifies the name of the key to use. By default this will be the property key alias. Used when
+     * the same transit key is used with different configurations. Such as:
+     * 
+     * <pre>
+     *     {@code
+     *     quarkus.vault.transit.key.foo1.name=foo
+     *     quarkus.vault.transit.key.foo1.key_version=1
+     *     quarkus.vault.transit.key.foo2.name=foo
+     *     quarkus.vault.transit.key.foo2.key_version=2
+     *     }
+     * </pre>
+     */
+    @ConfigItem
+    public Optional<String> name;
+
+    // sign
+
+    /**
+     * Set to true when the input is already hashed.
+     * applies to sign operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#prehashed">api documentation for prehashed</a>
+     */
+    @ConfigItem
+    public Optional<Boolean> prehashed;
+
+    /**
+     * When using a RSA key, specifies the RSA signature algorithm.
+     * applies to sign operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#signature_algorithm">api documentation for
+     *      signature_algorithm</a>
+     */
+    @ConfigItem
+    public Optional<String> signatureAlgorithm;
+
+    /**
+     * Specifies the way in which the signature should be marshaled.
+     * applies to sign operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#marshaling_algorithm">api documentation for
+     *      marshaling_algorithm</a>
+     */
+    @ConfigItem
+    public Optional<String> marshallingAlgorithm;
```

Comment:
I'd just drop it for now. Per the docs it is only relevant to the non RSA keys and it will also require us supporting a base64 url encoding, lets add when someone even needs it :-)

Reply:
I am not sure what you mean by requiring us to support a base64 url encoding.
I see that the document states: `Selecting this will also change the output encoding to URL-safe Base64 encoding instead of standard Base64-encoding`

I did a quick test with a key created as: `vault write transit/keys/jws type=ecdsa-p256` and signed a document with `quarkus.vault.transit.key.jws.marshalling-algorithm=jws`, and I got:
```
vault:v1:MEQCIEBfeYS/tM/m3Me7fLNGCOUFhC+94NgInQgNKdB3+3JvAiBft6hZx3DOASpXn/bLyhkyyP78Y+mxxCfZntfUiVkxWA==
```
the signature data is `URL-safe Base64` I suppose, but does it change anything for us? in all `sign`methods, we just assumed that signatures were `String`, just like we did for cypher texts in encryption methods. since we provide the result _as-is_, it does not change anything for us whether it is base64 or url-safe base64. This would be for the user to deal with.

It is one of those things (the same could be said for `signature_algorithm` and `prehashed` below) where I think it does not cost us a lot of work (a few unit tests, a few lines of code, some doc) to support those options now, even if that is not obvious for me that this is absolutely required at this point. and it is already implemented, in a non intrusive way (ie: in the config, not the api).

from what you are saying, it sounds like providing those options now is actually _not_ a good thing. I guess I am coming from the perspective that if it is in the api, I suppose somebody thought this would be needed, and if it is not too much work, then we may as well do it.

I do not feel too strong about it, so if you do, let me know and I will drop support for that option.

Reply:
@vsevel that is still a `base64` only I believe (2 trailing `=`), `base64url` encoding treats few characters differently, and also does not require a padding character like `=`. Please check one of the online decoders, I've no idea why Vault still has `=` in this case, but since at the Quarkus end it is only base64 encoding/decoding which is supported then if Vault does indeed support `base64url` option then there will be decoding errors at the either end in some cases.
So if you'd like to keep it then I believe we need to do more work around supporting a base64url encoding.
 
About the other properties like `prehashed` - this is OK to support them in general of course :-). But my point is, we need to have a good answer to anyone who would ask why we should enable this option instead of keeping it disabled, and if we don't have these answers yet then it is a good indication these properties have made it a bit early into the config API :-).

`signature_algorithm` is OK, I did not propose to remove it only suggested ti use a differemt default value but that was only a proposal, I'm OK with keeping the default one.

`prehashed` - I thought we actually can't have it supported unless we have a `hash` method ?

Reply:
so I dropped `marshalling_algorithm`. I am still not completely understanding the point about the base64. and since I do not see the problem, I am having a hard time figuring out a test that shows that it is actually working. so I will trust you on this.

based on your answer I left unchanged `prehashed` and `signature_algorithm`